Development practices/Design patterns.md

```diff
@@ -1,12 +1,74 @@
-- what are design patterns
-- list of design patterns in rails with links to external blogposts
-
-- previous chapters included:
-  - Adapters
-  - Data objects (Value objects)
-  - Decorators
-  - Form objects
-  - Null objects
-  - Policies
-  - Query objects
-  - Service objects
\ No newline at end of file
+
+## Classic design patterns
+
+Design patterns are typical solutions to common problems in software design.
+Each pattern is like a customizable blueprint used to solve a particular design problem in your code.
+They define a language that efficiently improves team communication.
+
+They can also be used to improve code readability and cleanness, as well as to simplify the design.
+
+[Refactoring.guru](https://refactoring.guru) is a great online place to learn more about them.
+
+## Rails patterns
+
+### Factory
+Class that has one creation method with a large conditional.
+
+#### Problem
+You have a method which instantiates a different type of object depending on a particular parameter.
+
+#### Solution
+Read more [here](https://www.sihui.io/design-pattern-factory/).
+
+### Form object
+Simplify record operations by keeping logic in them.
```

Comment:
I would refraze this somehow...
The old definition was somehow better for me.

Reply:
@d4be4st last changes: https://github.com/infinum/rails-handbook/pull/117/commits/b2ed21578f5a003fc562f5f30fefc32b17318ee6#diff-f5cea16a5f762555e1d1efbf4c751f96R24

Let me know if you agree with @melcha and we can merge it.